src/main/java/htsjdk/samtools/DuplicateScoringStrategy.java

```diff
@@ -128,7 +128,7 @@ public static int compare(final SAMRecord rec1, final SAMRecord rec2, final Scor
         int cmp;
 
         // always prefer paired over non-paired
-        if (rec1.getReadPairedFlag() != rec2.getReadPairedFlag()) return rec1.getReadPairedFlag() ? 1 : -1;
+        if (rec1.getReadPairedFlag() != rec2.getReadPairedFlag()) return rec1.getReadPairedFlag() ? -1 : 1;
```

Comment:
Will add a test. The only place it's used picard's MarkQueue.compare but it covers same logic correctly before it uses this method, so the bug is masked.